ipatests/test_integration/test_installation.py

```diff
@@ -1108,6 +1115,79 @@ def run_request(url, expected_stdout=None, expected_stderr=None):
                 expected_stdout=f'href="https://{self.master.hostname}/'
             )
 
+    @master_unistalled
+    def test_hostname_parameter(self):
+        """
+        Test for #2692 ipa-server-install ignores --hostname
+
+        https://pagure.io/freeipa/issue/2692
+        """
+        original_hostname = self.master.hostname
+        new_hostname = "new." + original_hostname
+        ip_address = self.master.ip
+        hosts = self.master.get_file_contents('/etc/hosts')
+        new_hosts = str(hosts) + "\n{}        {}".format(
+            ip_address, new_hostname)
+        self.master.put_file_contents('/etc/hosts', new_hosts)
+        result = tasks.install_master(self.master,
+                                      setup_dns=False,
+                                      extra_args=['--hostname',
+                                                  new_hostname])
+        try:
+            assert result.returncode == 0
+            hostname = self.master.run_command([
+                'hostname', '-f']).stdout_text.strip()
+            assert hostname == new_hostname
+        finally:
+            self.master.hostname = original_hostname
+            self.master.put_file_contents('/etc/hosts', hosts)
+
+    @master_unistalled
+    def test_ad_subpackage_dependency(self):
+        """
+        Test for #4011 ipa-server-install crashes
+        when AD subpackage is not installed
+
+        https://pagure.io/freeipa/issue/4011
+        """
+        package_name = "*ipa-server-trust-ad"
+        tasks.uninstall_packages(self.master, [package_name])
+        result = tasks.install_master(self.master)
+        assert result.returncode == 0
+
+    @master_unistalled
+    def test_backup_of_cs_cfg_should_be_created(self):
+        """
+        Test for #4166 Backup CS.cfg before modifying it
+
+        https://pagure.io/freeipa/issue/4166
+        """
+        bcp_location = paths.CA_CS_CFG_PATH + ".ipabkp"
+        tasks.install_master(self.master)
+        ipaserver_install_log = str(self.master.get_file_contents(
+            paths.IPASERVER_INSTALL_LOG))
+        assert "backing up CS.cfg" in ipaserver_install_log
+        try:
+            self.master.get_file_contents(bcp_location)
```

Comment:
It will be nice to also check that contents of the backup are really equal to what was in the file before the installation.

Reply:
The file is not available before the installation, so unless I install twice, I can't get the "original" file.